projects/capnproto/fuzz-run.c++

```diff
@@ -0,0 +1,71 @@
+#include "test-util.h"
+#include <kj/main.h>
+#include "serialize.h"
+#include <capnp/test.capnp.h>
+#include <unistd.h>
+
+namespace capnp {
+    namespace _ {
+        namespace {
+
+  class ParseErrorCatcher: public kj::ExceptionCallback {
+  public:
+    ParseErrorCatcher(kj::ProcessContext& context): context(context) {}
+    ~ParseErrorCatcher() noexcept(false) {
+      if (!unwindDetector.isUnwinding()) {
+        KJ_IF_MAYBE(e, exception) {
+          context.error(kj::str(
+              "*** ERROR CONVERTING PREVIOUS MESSAGE ***\n"
+              "The following error occurred while converting the message above.\n"
+              "This probably means the input data is invalid/corrupted.\n",
+              "Exception description: ", e->getDescription(), "\n"
+              "Code location: ", e->getFile(), ":", e->getLine(), "\n"
+              "*** END ERROR ***"));
+        }
+      }
+    }
+
+    void onRecoverableException(kj::Exception&& e) {
+      // Only capture the first exception, on the assumption that later exceptions are probably
+      // just cascading problems.
+      if (exception == nullptr) {
+        exception = kj::mv(e);
+      }
+    }
+
+  private:
+    kj::ProcessContext& context;
+    kj::Maybe<kj::Exception> exception;
+    kj::UnwindDetector unwindDetector;
+  };
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* Data, size_t Size)
+{
+    kj::TopLevelProcessContext context("fuzz");
```

Comment:
You don't need this if you aren't using KJ main.

Reply:
Indeed, but KJ main defines `main` function, and we want to use libFuzzer `main` function...
So, do you see another way around this ?

Reply:
No no, that's not what I meant. I'm saying there's no reason to create a `TopLevelProcessContext` when you aren't using KJ main. You aren't, so you can delete this line.

Reply:
Ok thanks, I will